app/src/lib/stores/helpers/infer-compare-to-branch.ts

```diff
@@ -0,0 +1,153 @@
+        branches,
+        needsAName.repository.gitHubRepository!
+      )
+    case 'fork':
+      return _getDefaultBranchOfFork(
+        branches,
+        needsAName.repository,
+        needsAName.gitHubRepository,
+        needsAName.currentBranch
+      )
+    default:
+      return _getMasterBranch(branches)
+  }
+}
+
+/**
+ * Do **not** call this method; use `inferComparisonBranch` instead
+ *
+ * Returns `master`, the default branch of a Git repository
+ *
+ * @param branches The list of all branches for the repository
+ */
+export function _getMasterBranch(
+  branches: ReadonlyArray<Branch>
+): Branch | null {
+  return branches.find(b => b.name === 'master') || null
+}
+
+/**
+ * Do **not** call this method; use `inferComparisonBranch` instead
+ *
+ * Returns the default branch of the GitHub repository
+ *
+ * @param branches The list of all branches for the repository
+ * @param ghRepository The repository the branch belongs to
+ */
+export function _getDeafultBranchOfGithubRepo(
+  branches: ReadonlyArray<Branch>,
+  ghRepository: GitHubRepository
+): Branch | null {
+  return branches.find(b => b.name === ghRepository.defaultBranch) || null
+}
+
+/**
+ * Do **not** call this method; use `inferComparisonBranch` instead
+ *
+ * Returns the base branch of the given pull request.
+ *
+ * @param branches The list of all branches for the repository
+ * @param pr The pull request to use for finding the branch
+ */
+export function _getFeatureBranchOfPullRequest(
+  branches: ReadonlyArray<Branch>,
+  pr: PullRequest
+): Branch | null {
+  return branches.find(b => b.name === pr.base.ref) || null
+}
+
+/**
+ * Do **not** call this method; use `inferComparisonBranch` instead
+ *
+ * Returns the default branch of the fork if it's ahead of `currentBranch`.
+ * Otherwise, the default branch of the parent is returned.
+ *
+ * @param branches The list of all branches for the repository
+ * @param repository The repository the branch belongs to
+ * @param ghRepository
+ * @param currentBranch The branch we want the parent of
+ * @param getAheadBehind Callback function used to compute ahead/behind
+ */
+export async function _getDefaultBranchOfFork(
+  branches: ReadonlyArray<Branch>,
+  repository: Repository,
+  ghRepository: GitHubRepository,
+  currentBranch: Branch,
+  mockingAheadBehind?: (range: string) => IAheadBehind
+): Promise<Branch | null> {
+  const defaultBranchName = ghRepository.defaultBranch
+
+  if (defaultBranchName === null) {
+    // Fall back to master
+    return _getMasterBranch(branches)
+  }
+
+  const defaultBranch = _getDeafultBranchOfGithubRepo(branches, ghRepository)
+
+  if (defaultBranch === null) {
+    // Fall back to master
+    return _getMasterBranch(branches)
+  }
+
+  const range = revRange(currentBranch.tip.sha, defaultBranch.tip.sha)
+
+  // use the mock if it's passed - for testing only
+  const aheadBehind =
+    (mockingAheadBehind && mockingAheadBehind(range)) ||
```

Comment:
I need to be able to test this function by controlling what this line returns. Since this function is meant to be private, I felt it would be OK to pass an optional argument, `mockingAheadBehind`, to this function and call it in this way. I'm open to other ideas for mocking the behavior of the `getAheadBehind` function.